docs/content/concepts/ops-jobs-graphs/ops.mdx

```diff
@@ -1,42 +1,41 @@
+| <PyObject object="In" />                | An input to to an op. Defined on the `ins` argument to the <PyObject object="op" decorator/> decorator.                                                    |
+| <PyObject object="Out" />               | An output of an op. Defined on the `out` argument to the <PyObject object="op" displayText="op" decorator/> decorator.                                |
+| <PyObject object="OpExecutionContext"/> | An object exposing Dagster system APIs for resource access, logging, and more. Can be injected into an op by specifying `context` as the first argument of the compute function.                           |
+| <PyObject object="OpDefinition"  />     | Class for ops. You will rarely want to instantiate this class directly. Instead, you should use the <PyObject object="op" decorator />. |
 
 ## Overview
 
-Ops are used to define computations. Ops can later be assembled into [Jobs](/concepts/ops-jobs-graphs/jobs-graphs). Ops generally perform one specific action and are used for batch computations. For example, you can use an op to:
+Ops are Dagster's atomic unit of computation. Individual ops should perform relatively simple tasks. Collections of ops can then be assembled into [Jobs or Graphs](/concepts/ops-jobs-graphs/jobs-graphs) to perform more complex tasks. Some examples of tasks appropriate for a single op:
 
-- Derive a data set from some other data sets.
+- Add computed columns to a dataframe.
 - Execute a database query.
 - Initiate a Spark job in a remote cluster.
 - Query an API and store the result in a data warehouse.
 - Send an email or Slack message.
 
-Depending on how you execute a job, Dagster can run all the ops in the job within the same process or isolate them in different processes.
+The op as computational unit enables many useful features for data orchestration:
 
-Ops have several important properties:
-
-- **Inputs and Outputs**: Ops have defined [inputs and outputs](#inputs-and-outputs), which can be optionally typed. These types are validated at runtime.
-- **Configurable**: Ops can be configured, using a strongly typed [configuration system](/concepts/configuration/config-schema).
-- **Emit an Event Stream**: Ops can emit a stream of structured events, such as AssetMaterializations. These events can be viewed in [Dagit](/concepts/dagit/dagit), Dagster's UI tool.
-- **Individually testable**: See the [testing page](/concepts/testing) for more detail.
+- **Flexible execution strategies**: Data pipelines are frequently developed locally and later deployed to production. Ops represent sealed units of logic independent of pipeline execution strategy, making the transition from development to production painless. Collections of ops (i.e. [Jobs](/concepts/ops-jobs-graphs/jobs-graphs)) can be passed to an appropriate [Executor](/deployment/executors) for single-process execution or distribution across a cluster.
```

Comment:
"passed to an appropriate Executor" isn't exactly accurate. Each job has an executor - i.e. the executor is determined at job definition time.

Reply:
I like the explanation here! Maybe rephrase this bullet "Flexible execution strategies" to "Business logic separation" or "Environment isolation"? bc besides executors, an op can also isolate the biz logic from resources across different execution environments.

Reply:
Thx for pointer on executors. My preferred solution here relates to my point above about graphs.

In several places I've seen collections of ops described as jobs, or references to connecting ops to form a job. For instance, the headlines for Op and Job concept pages in current docs say:

- "Multiple ops can be connected to create a Job."
- "A job is an executable graph of ops"

I understand a decision was made to emphasize jobs over graphs at some point, presumably because it introduces fewer abstractions when you're trying to explain how to get something done.

But I think this has resulted in elisions that are ultimately costly to the user, because it creates confusion about Dagster works. IMO it is better to always describe collections of connected ops  as capital-g "Graphs", and emphasize that a Graph is a component of a Job-- Ops are computational units, they can be wired together into Graphs, and a Job binds configuration/resources and an executor (and maybe some other stuff) with a Graph. Use of the @job decorator is just a way of creating an anonymous Graph, analogous to an inline lambda in Python.

Following this approach, I'd edit that last sentence like so:

>Collections of ops (i.e. [Graphs](/concepts/ops-jobs-graphs/jobs-graphs)) are bound via [Jobs](/concepts/ops-jobs-graphs/jobs-graphs) to an appropriate [Executor](/deployment/executors) for single-process execution or distribution across a cluster.